apiserver/facades/controller/undertaker/state.go

```diff
@@ -44,12 +44,17 @@ type State interface {
 
 type stateShim struct {
 	*state.State
+	model *state.Model
```

Comment:
This is really the wrong pattern. stateShim wraps State. We have a separate modelShim to wrap Model. The code that uses this stuff needs to use the required semantics. eg that probably means getting a model off state and then calling ModelConfig() on the model.